### DISCUSSION TITLE
Get list of mentions with suggestion extension

### DISCUSSION THREAD

**Discussion by amirhbakan** [ueberdosis/tiptap#990] [General Questions]
Hi, I have trouble with having the list of mentions in the editor. I can add them to my data with onEnter event but I don't have any event on mention deletion to delete the deleted mentions from my data, how can I do that?
or how can I get the list of mentions from the editor?

**Comment by sipec** [comment]
I had the same issue, and I settled on just parsing the JSON myself:
```javascript
function parseMentions(data) {
  const mentions = (data.content || []).flatMap(parseMentions)
  if (data.type === 'mention') {
    mentions.push(data.attrs.id)
  }
  return mentions
}

// how to use:
parseMentions(editor.getJSON())
```
Basically this just uses recursion to depth-first-search for all the mentions, returning them as a list of strings

**Reply by codellyson** [reply]
you are a life saver

**Reply by ScottSavarie** [reply]
Added this extra line to dedupe multiple entries. Thanks for this one! 

```
function parseMentions(data) {
  const mentions = (data.content || []).flatMap(parseMentions)
  if (data.type === 'mention') {
    mentions.push(data.attrs.id)
  }

  const uniqueMentions = [...new Set(mentions)]

  return uniqueMentions
}
```

**Comment by joe-pomelo** [comment]
An old issue, but here is related answer on extracting mentions from `JSONContent`: 
https://github.com/ueberdosis/tiptap/discussions/2569#discussioncomment-2235356
